### ISSUE TITLE
Change `/unique_address` callback endpoint to something more meaningful

### ISSUE THREAD

**Issue by lijamie98** [stellar/java-stellar-anchor-sdk#557]

**Comment by JakeUrban** [comment]
Thanks for creating the issue @lijamie98, here are my thoughts.

The "unique" piece of this endpoint name can be replaced with a `POST` method. `POST` requests are by definition meant to create a unique resource.

In fact, I think we can mirror Circle's endpoint, using `POST /addresses`.

**Comment by marcelosalloum** [comment]
To be a little more precise with the data returned from that endpoint's response, I'd go with something like `POST /deposit_info`, since we're retrieving not only an address but also a memo.

This also provides more information about what the endpoint is used for. For example, when you create an address, it may not be very clear what it will be used for. When you create the deposit info though, it's more explicit that it will be used for deposits.

**Comment by JakeUrban** [comment]
I agree that `/deposit_info` is more accurate @marcelosalloum, but personally I'd like to use the single-word `POST /addresses`. BTW, have we decided if we'd use underscores vs. camel-case (`/deposit_info` vs. `/depositInfo`)?
